hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/service/model/ContractDebugParameters.java

```
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.web3.service.model;
+
+import com.hedera.mirror.web3.evm.contracts.execution.traceability.TracerType;
+import com.hedera.mirror.web3.viewmodel.BlockType;
+import com.hedera.node.app.service.evm.store.models.HederaEvmAccount;
+import lombok.Builder;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+
+@Value
+@Builder(toBuilder = true)
+@RequiredArgsConstructor
+public class ContractDebugParameters implements CallServiceParameters {
+    BlockType block;
+    Bytes callData;
+    CallType callType = CallType.ETH_DEBUG_TRACE_TRANSACTION;
+    long consensusTimestamp;
+    long gas;
+    boolean isEstimate = false;
+    boolean isStatic = false;
+    Address receiver;
+    HederaEvmAccount sender;
+    TracerType tracerType = TracerType.OPCODE;
+    long value;
+}
```

```suggestion
public class ContractDebugParameters implements CallServiceParameters {

     @NotNull
     BlockType block;

     @NotNull
     Bytes callData;

     @NotNull
     CallType callType = CallType.ETH_DEBUG_TRACE_TRANSACTION;

     @Positive
     long consensusTimestamp;

     @PositiveOrZero
     long gas;

     @AssertFalse
     boolean isEstimate = false;

     @AssertFalse
     boolean isStatic = false;

     @NotNull
     Address receiver;

     @NotNull
     HederaEvmAccount sender;

     @NotNull
     TracerType tracerType = TracerType.OPCODE;

     @PositiveOrZero
     long value;
 }
```

---

It's quite inefficient and redundant to validate user input twice just because there was a temporary mapping bug in a PR. I think the main problem here is that you are doing integration tests in a separate PR so you're not catching things in this one.

---

We are not validating user input here, but it’s more like a validation which checks that the debug service parameters that we have built are valid. 

If the validation fails here, it means that we have a bug in the logic and it’s an early sign that something is not okay. If we remove this validation the bug wont be caught anywhere and the method will silently continue despite receiving invalid input. It’s a good practice to have such checks instead of completely relying on tests to cover all edge cases.